app/assets/javascripts/autocomplete.coffee

```diff
@@ -0,0 +1,54 @@
+delay = do ->
+  timer = 0
+  (callback, ms) ->
+    clearTimeout timer
+    timer = setTimeout(callback, ms)
+    return
+
+update_textfield = (list_element) ->
+  $(list_element).removeClass('suggestion-focused')
+  $('.js-autocomplete-textfield').val($(list_element).find('.res-name')[0].innerHTML)
```

Comment:
:+1: 

This is a possible case, I would change the name if I am not satisfied with it :stuck_out_tongue:

What do you think about adding a hidden `input` with `repository_id` as `value` and we can use the value if it's present?